[EN] update Cloud Computing (#2118)
cncf/glossary: content/en/cloud-computing.md

tags: ["infrastructure", "fundamental", ""]
---

> Cloud computing is a method of providing computing resources such as CPU power, storage, and network capabilities via the internet. It allows users to leverage remote computing power located in distant data centers. There are two types: private cloud, dedicated to a single organization, and public cloud, open for general use.

Reviewer: ```suggestion
Cloud computing offers CPU power, storage, and network capabilities over the internet, enabling scalable and flexible access to resources across global data centers. It spans private clouds, dedicated to single organizations for security and control, and public clouds, open for widespread use, optimizing cost and scalability.
```

## Problem it addresses

> Historically, organizations looking to increase their computing capacity faced a dilemma: either invest in new facilities for their servers and networks or upgrade existing ones. This process was time-consuming and resource-intensive.

Reviewer: ```suggestion
Traditionally, organizations needing more computing capacity had to choose between costly investments in new server facilities or upgrades to existing infrastructure—a slow and resource-heavy process.
```

## How it helps

> Cloud computing addresses this by enabling organizations to rent computing resources as needed, without the burden of managing physical infrastructure. This approach offers two main advantages. Firstly, it frees organizations to concentrate on their core business, eliminating delays and costs associated with setting up new infrastructure. Secondly, it provides the flexibility to [scale](https://github.com/ronitblenz/glossary/blob/cloud_computing/content/en/scalability.md) resources up or down based on demand, aligning infrastructure needs closely with business requirements. Thus, cloud computing offers a practical solution for organizations to access the amount of infrastructure they need, when they need it, without overcommitting resources.

Reviewer: ```suggestion
Organizations can use cloud computing to rent computing resources on demand without the burden of managing physical infrastructure. This strategy has two main advantages:
- It removes the delays and expenses of establishing new infrastructure so organizations can concentrate on their core business.
- Organizations can [scale](https://github.com/ronitblenz/glossary/blob/cloud_computing/content/en/scalability.md) their resources up or down based on demand, aligning infrastructure with business needs.
Thus, cloud computing provides an efficient way for organizations to access necessary infrastructure flexibly and economically without excess commitment.
```

tags: ["infrastructure", "fundamental", ""]
---

> Cloud computing offers CPU power, storage, and network capabilities over the internet, enabling scalable and flexible access to resources across global data centers. It spans private clouds, dedicated to single organizations for security and control, and public clouds, open for widespread use, optimizing cost and scalability.

Reviewer 2: ```suggestion
Cloud computing offers CPU power, storage, and network capabilities over the internet, 
enabling scalable and flexible access to resources across global data centers. 
It spans private clouds, dedicated to single organizations for security and control, 
and public clouds, open for widespread use, optimizing cost and scalability.
```

## Problem it addresses

> Traditionally, organizations needing more computing capacity had to choose between costly investments in new server facilities or upgrades to existing infrastructure—a slow and resource-heavy process.

Reviewer 2: ```suggestion
Traditionally, organizations needing more computing capacity had to choose between costly investments in new server facilities or upgrades to existing infrastructure, a slow and resource-heavy process.
```

## How it helps

> Organizations can use cloud computing to rent computing resources on demand without the burden of managing physical infrastructure. This strategy has two main advantages:

Reviewer 2: ```suggestion
Organizations can use cloud computing to rent computing resources on demand without the burden of managing physical infrastructure. 
This strategy has two main advantages:
```

## What it is

Cloud computing offers compute resources like CPU, network, and disk capabilities on-demand over the internet, allowing users to access and use computing power in a remote physical location. 

> It extends beyond merely providing resources and we generally differentiate between private and public cloud. These models differ based on the usage and accessibility of the cloud infrastructure.

Reviewer 3: Not sure I understand. What does it provide that extends beyond providing resources?

Author: Providing Resources in the sense, apart from providing computational needs, we have models based on our requirements. Basically we can cover specific use cases. I have quoted more on the Private, public and hybrid cloud solutions which was proposed in the issue itself.

Reviewer 3: Hmm...I think in private, public, and hybrid clouds, you are still getting resources. The question is, who do you share those resources with? I would say, at the core, they offer virtualized resources. Cloud providers also provide PaaS (e.g., AKS), which I belive is more of an extension of their cloud computing services. 

But I may be wrong. Let's ask some of our technical team to chime in. @nate-double-u @iamNoah1 @castrojo, any thoughts?

> In a private cloud, the infrastructure is exclusively dedicated to a single organization, and it can exist both on-premise (within the organization's physical location) or off-premise (hosted by a third-party service provider). This setup offers more control over the infrastructure, data security, and regulatory compliance, and can be customized to meet specific business needs.

Reviewer 2: ```suggestion
In a private cloud, the infrastructure is exclusively dedicated to a single organization, 
and it can exist both on-premise (within the organization's physical location) or off-premise (hosted by a third-party service provider). 
This setup offers more control over the infrastructure, data security, and regulatory compliance, 
and can be customized to meet specific business needs.
```

> On the other hand, a public cloud is a shared infrastructure available for open public services, where resources are owned and operated by third-party cloud service providers. 

Reviewer 2: ```suggestion
On the other hand, a public cloud is a shared infrastructure available for open public services, 
where resources are owned and operated by third-party cloud service providers. 
```

> Additionally, a hybrid cloud model combines aspects of both private and public clouds. This model enables businesses to leverage the advantages of both deployment models by integrating their private cloud with the public cloud. This combination can be used for creating a balanced, flexible, and cost-effective cloud environment.

Reviewer 2: ```suggestion
Additionally, a hybrid cloud model combines aspects of both private and public clouds. 
This model enables businesses to leverage the advantages of both deployment models by integrating their private cloud with the public cloud. 
This combination can be used for creating a balanced, flexible, and cost-effective cloud environment.
```

> Therefore, the distinction among cloud computing models isn't solely about whether the infrastructure is private or public. It also involves the location of the infrastructure and the customization level it can offer, as well as the possibility to create a hybrid solution that maximizes the benefits of both environments.

Reviewer 2: ```suggestion
Therefore, the distinction among cloud computing models isn't solely about whether the infrastructure is private or public. 
It also involves the location of the infrastructure and the customization level it can offer, 
as well as the possibility to create a hybrid solution that maximizes the benefits of both environments.
```

## What it is

Cloud computing offers compute resources like CPU, network, and disk capabilities on-demand over the internet, allowing users to access and use computing power in a remote physical location. 

> It extends beyond merely providing resources and we generally differentiate between private and public cloud. These models differ based on the usage and accessibility of the cloud infrastructure.

Reviewer 2: ```suggestion
It extends beyond merely providing resources and we generally differentiate between private and public cloud. 
These models differ based on the usage and accessibility of the cloud infrastructure.
```

## What it is

Cloud computing offers compute resources like CPU, network, and disk capabilities on-demand over the internet, allowing users to access and use computing power in a remote physical location. 

> It extends beyond merely providing resources and we generally differentiate between private and public cloud. 

Reviewer 4: I wonder if we should expand the cloud computing definition to include the different service models of cloud computing too - IaaS (Infrastructure as a Service), PaaS (Platform as a Service), and SaaS (Software as a Service).

This may be out of scope for this definition tho. @iamNoah1, thoughts?

Reviewer 3: Hmm...this section is already very long, and we want to be sure to stick to our [minimal viable definition](https://glossary.cncf.io/style-guide/#minimum-viable-definition) goal. We had separate definitions or issues to create them for IaaS, PaaS, and SaaS but decided not to include them because they were not cloud native specific or necessary to understand cloud native concepts. 

Going back to the minimal viable definition, what are the most important things readers need to understand about this concept to make sense of it when reading about it in other resources? And let's stick to that.

Reviewer 2: I personally like the idea of at least mentioning that when we talk about cloud computing we have different abstractions like Paas, IaaS, SaaS, and even FaaS if you want. On the other side and correct me if I am wrong @CathPag @seokho-son @jihoon-seo at some point we decided to want to include XaaS, isn't it?! 
So maybe as a suggestion, let's say no to XaaS for this PR and now. Let's talk about this topic in the next glossary meeting, and then adjust accordingly.

Author: So, I will wait till the maintainers and the core team decides to include PaaS, IaaS, SaaS and FaaS. Drop a text here for confirmation after the meeting! @iamNoah1 @CathPag

Reviewer 3: We will skip it for now, so closing this comment.

These models differ based on the usage and accessibility of the cloud infrastructure.

In a private cloud, the infrastructure is exclusively dedicated to a single organization, 

> and it can exist both on-premise (within the organization's physical location) or off-premise (hosted by a third-party service provider). 

Reviewer 5: The meaning of `premise` and `premises` are totally different.

```suggestion
and it can exist both on-premises (within the organization's physical location) or off-premises (hosted by a third-party service provider). 
```

This setup offers more control over the infrastructure, data security, and regulatory compliance, 
and can be customized to meet specific business needs.

On the other hand, a public cloud is a shared infrastructure available for open public services, 

> where resources are owned and operated by third-party cloud service providers. 

Reviewer 4: While it's true that the infrastructure is shared, it might be misunderstood that the data is shared as well. It's important to clarify that while resources are shared in a public cloud, each user's data and applications remain isolated from others.

Author: Indeed ... Lets update that

Additionally, a hybrid cloud model combines aspects of both private and public clouds. 
This model enables businesses to leverage the advantages of both deployment models by integrating their private cloud with the public cloud. 

> This combination can be used for creating a balanced, flexible, and cost-effective cloud environment.

Reviewer 4: In addition to the hybrid cloud model, there's also the concept of multi-cloud, where an organization uses multiple public cloud services to meet specific workload needs or for redundancy, flexibility, and avoiding vendor lock-in.

Author: Will add that as well!
